src/go/configgenerator/listener_generator.go

```diff
@@ -623,21 +623,10 @@ func makeServiceControlFilter(serviceInfo *sc.ServiceInfo) *hcmpb.HttpFilter {
 			},
 		}
 	} else {
-		// Use access token from fetched the Instance Metadata Server to talk to Service Controller
-		switch serviceInfo.AccessToken.TokenType.(type) {
-		case *commonpb.AccessToken_RemoteToken:
-			filterConfig.AccessToken = &scpb.FilterConfig_ImdsToken{
-				ImdsToken: serviceInfo.AccessToken.GetRemoteToken(),
-			}
-			break
-		case *commonpb.AccessToken_ServiceAccountSecret:
-			filterConfig.AccessToken = &scpb.FilterConfig_ServiceAccountSecret{
-				ServiceAccountSecret: serviceInfo.AccessToken.GetServiceAccountSecret(),
-			}
-			break
-		default:
-			break
+		filterConfig.AccessToken = &scpb.FilterConfig_ImdsToken{
+			ImdsToken: serviceInfo.AccessToken.GetRemoteToken(),
```

Comment:
`GetRemoteTokenUri` or `GetAccessTokenUri`

Otherwise it sounds like you're passing the actual token value.

Reply:
Nevermind, this is part of the proto file. Then it's fine, no need to change the name unless you think it's worth it.